### PR TITLE
feat: Filter by profile type

### LINKED ISSUE #822: Searching by Community or Personal profile
### Description

I think we can improve search function without changing much of the Search bar UI, we can use something like type:community|personal, just like github search functionality __is:open__, etc... .Also, to make the search work either we have to make two seperate folders for containing personal profiles and other with community profile __OR__ we can decide to have a better naming syntax for the json data files like community profiles starting with __comm-username.json__ or any other suggested by the community.

### Screenshots

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by nathgoutam93**
## Fixes Issue

Closes #822

## Check List

- [X] My code follows the code style of this project.
- [ ] My change requires changes to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] All new and existing tests passed.
- [X] This PR does not contain plagiarized content.
- [X] The title of my pull request is a short description of the requested changes.

## Screenshots

https://user-images.githubusercontent.com/91387097/148654527-9f41549c-8628-4a8d-b01c-d5cf4e65a729.mp4

<a href="https://gitpod.io/#https://github.com/EddieHubCommunity/LinkFree/pull/937"></a>

### CODE REVIEW

**Review by emmalearnscode — CHANGES_REQUESTED** [review]
I think 'all' would be a better word to use instead of 'misc' as you are showing all profiles by default. Misc infers some sort of randomness to it which there isn't.

**Comment by nathgoutam93** [comment]
> I think 'all' would be a better word to use instead of 'misc' as you are showing all profiles by default. Misc infers some sort of randomness to it which there isn't.

Indeed, I wasn't sure what word to use

**Review by krishguptadev — COMMENTED** [review]
We have got the wording 'personal' and 'community' already, perhaps you can bring those to the use instead of misc?

**Comment by nathgoutam93** [comment]
> We have got the wording 'personal' and 'community' already, perhaps you can bring those to the use instead of misc?

I'm already using them , but as a default showing misc profiles both personal and community . But as @emmalearnscode  suggested I'll change the "misc" with "All"

**Review by emmalearnscode — CHANGES_REQUESTED** [review]
Since we are using prime react, could you change the dropdown to this one so that we keep consistent styles? https://www.primefaces.org/primereact/dropdown/

Thank you :) (And apologies I didn't notice it before)

**Comment by nathgoutam93** [comment]
> Since we are using prime react, could you change the dropdown to this one so that we keep consistent styles? https://www.primefaces.org/primereact/dropdown/
> 
> Thank you :) (And apologies I didn't notice it before)

yup I tried but the problem is that , as primereact had the dependency of prime icons the icons for some components stopeed showing as they do not exist anymore after removal of primeicons . I also tried to use react icons in the component but when i look into the source of drop down component it do not take icons directly instead it takes css class. I'll try to come up with a solution if not then maybe we will need to add prime icons back as it is a dependency of preimereact

**Comment by emmalearnscode** [comment]
> as primereact had the dependency of prime icons

If the components depend on prime icons then they should be installed back into the project, otherwise we will run into the same issue every time we try to use a prime react component. It shouldn't be a problem to have both icon packages there as far as I know.

**Comment by nathgoutam93** [comment]
> > as primereact had the dependency of prime icons
> 
> If the components depend on prime icons then they should be installed back into the project, otherwise we will run into the same issue every time we try to use a prime react component. It shouldn't be a problem to have both icon packages there as far as I know.

should I install primeicons in this PR or an issue should be raised?

**Comment by nathgoutam93** [comment]
firefox_xhpFiYKTAX

as you can see in screenshot the drop down icon is not working

**Comment by emmalearnscode** [comment]
Add it with this PR as it is something you need to get the dropdown working properly

**Review by emmalearnscode — APPROVED** [review]
Looks like it's working well to me. Let's just wait for a couple more reviews to make sure nothing has been missed :)

**Review by Panquesito7 — APPROVED** [review]
LGTM 🚀 Thank you for your contribution! 😄👍

File: src/Components/Home/filterProfileType.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { Dropdown } from 'primereact/dropdown'
+
+export default function ProfileTypeFilter({ profileType, typeHandler }) {
+  const profileTypes = [
+    { label: 'Personal', value: 'personal' },
+    { label: 'All', value: 'all' },
+    { label: 'Community', value: 'community' },
```
Comment (schmelto) [inline src/Components/Home/filterProfileType.js:9]: I would prefer to make the selection in alphabetical order. 

image

```suggestion
    { label: 'All', value: 'all' },
    { label: 'Community', value: 'community' },
    { label: 'Personal', value: 'personal' }
```

**Review by schmelto — CHANGES_REQUESTED** [review]

File: src/Components/Home/filterProfileType.js (hunk 1 above)
Comment (nathgoutam93) [reply src/Components/Home/filterProfileType.js:9]: firefox_h8Ha0XXI8K

I think length order will look more neat ,what you say?

Comment (schmelto) [reply src/Components/Home/filterProfileType.js:9]: I would stick to alphabetical but I leave it for discussion.
But I think we can convert this to a new issue and merge your changes.

**Review by schmelto — APPROVED** [review]

**Pull request merged**